tensordict/tensordict.py

```diff
@@ -111,11 +111,13 @@ class KeyedJaggedTensor:
 
 
 def is_tensordict(datatype: type | Any) -> bool:
+    from tensordict.prototype import is_tensorclass
+
     return (
         issubclass(datatype, TensorDictBase)
         if isinstance(datatype, type)
         else isinstance(datatype, TensorDictBase)
-    )
+    ) or is_tensorclass(datatype)
```

Comment:
Seems a bit surprising to me that `is_tensordict(some_tensorclass)` would return `True`. What's the motivation for this?

Reply:
with this PR tensorclass becomes interchangeable with tensordict in many cases. We could rename the check though, I agree it's confusing.

Reply:
To elaborate: every time we check if something is a tensordict, it is to propagate a method to its fields. The same is now done if it's a tensorclass.

Reply:
@tcbegley I added a docstring. I'm cheating a bit, `is_tensordict` stands for "if from the tensordict library" 😅